File: DSA/Cpp/test/dsa/lib/math/ds/anagram_test.cpp
#include <dsa/lib/math/ds/anagram.hpp>
#include "ds_test.h"
namespace dsa::lib::math::ds {
namespace {

class AnagramTest : public DsTest {
 protected:
    static bool isAnagram(const char *s, const char *t) {
        return Anagram::isAnagram(s, t);
    }
};
TEST_F(AnagramTest, case1) {
    bool actual = isAnagram("anagram", "nagaram");
    EXPECT_TRUE(actual);
}
TEST_F(AnagramTest, case2) {
    bool actual = isAnagram("rat", "car");
    EXPECT_FALSE(actual);
}

}
}

File: DSA/Java/test/dsa/lib/math/ds/AnagramTest.java
package dsa.lib.math.ds;

import org.junit.Test;

import static org.junit.Assert.*;

public class AnagramTest {

    @Test
    public void case1() {
        assertTrue(isAnagram("anagram", "nagaram"));
        assertFalse(isAnagram("rat", "car"));
    }

    private boolean isAnagram(String s, String t) {
        return new Anagram().isAnagram(s, t);
    }

}